src/tlo/methods/healthsystem.py

```diff
@@ -337,6 +343,45 @@ def make_appt_footprint(self, dict_of_appts):
             "values"
         )
 
+    def get_equip_item_code_from_item_name(self, lookup_df: pd.DataFrame, equip_item_name: str) -> int:
+        """Helper function to provide the equip_item_code (an int) when provided with the equip_item_name of the item"""
+        return int(pd.unique(lookup_df.loc[lookup_df["Equip_Item"] == equip_item_name, "Equip_Code"])[0])
+
+    def set_equipment_essential_to_run_event(self, set_of_equip):
+        """Helper function to set essential equipment.
+
+        Should be passed a set of equipment items names (strings) or an empty set.
+        """
+        # Set EQUIPMENT if the given set_of_equip in correct format, ie a set of strings or an empty set
+        if not isinstance(set_of_equip, set) or any(not isinstance(item, str) for item in set_of_equip):
+            raise ValueError(
+                "Argument to set_equipment_essential_to_run_event should be an empty set or a set of strings of "
+                "equipment item names from ResourceFile_Equipment.csv."
+            )
+
+        self.ESSENTIAL_EQUIPMENT  = set_of_equip
+
+    def add_equipment(self, set_of_equip):
+        """Helper function to update equipment.
+
+        Should be passed a set of equipment item names (strings).
+        """
+        # Update EQUIPMENT if the given set_of_equip in correct format, ie a non-empty set of strings
+        if not isinstance(set_of_equip, set) or any(not isinstance(item, str) for item in set_of_equip) or \
+           (set_of_equip in [set(), None, {''}]):
+            raise ValueError(
+                "Argument to add_equipment should be a non-empty set of strings of "
+                "equipment item names from ResourceFile_Equipment.csv."
+            )
+        # from the set of equip item names create a set of item codes
+        # this function is calling parameters from this
+        equip_codes = set(self.get_equip_item_code_from_item_name(
+            self.sim.modules['HealthSystem'].parameters['equip_item_and_package_code_lookups'], item_name
+        ) for item_name in set_of_equip
+        )
```

Comment:
```suggestion
        equip_codes = set(
            self.get_equip_item_code_from_item_name(
                self.sim.modules['HealthSystem'].parameters['equip_item_and_package_code_lookups'], 
                item_name
            ) for item_name in set_of_equip
        )
```
Having the nested parentheses at the same indentation level makes it a bit difficult to see the block structure here.

Also is there a reason to have `self.sim.modules['HealthSystem'].parameters['equip_item_and_package_code_lookups']` passed as an argument to `get_equip_item_code_from_item_name` method rather than just directly accessed within that method using `self` argument, as I think that would help make the code a bit more easily readable here?

Reply:
I think `self` directs to the HSI event here not to the HealthSystem module, as if I use only `self.parameters['equip_item_and_package_code_lookups']` instead of `self.sim.modules['HealthSystem'].parameters['equip_item_and_package_code_lookups']` I get an `AttributeError: 'HSI_Contraception_FamilyPlanningAppt' object has no attribute 'parameters'`

Reply:
Sorry here I meant as the `self` argument in the `get_equip_item_code_from_item_name` method could also be used to access `self.sim.modules['HealthSystem'].parameters['equip_item_and_package_code_lookups']` I don't think you need to have it passed in to the method at all so you could call it as `self.get_equip_item_code_from_item_name(item_name)`

Reply:
Ah, got it now. Yes, that makes sense. Done.